calendar.rb

```diff
@@ -0,0 +1,203 @@
+    year = time.year
+    show_calendar(month, year)
+  end
+
+  def any_month_calendar
+    month = 0
+    year = 0
+    loop do
+      begin
+        loop do
+          print "Enter year: "
+          year = Integer(gets)
+          if year != nil && year != "" && year >= 1500 && year <= 3000
+            break
+          else
+            puts "Year can be between 1500 to 3000".yellow
+          end
+        end
+        loop do
+          print "Enter month: "
+          month = Integer(gets)
+          if month != nil && month != "" && month >= 1 && month <= 12
+            break
+          else
+            puts "Month can be between 1 to 12".yellow
+          end
+        end
+        if Date.valid_date?(year,month,1)
+          show_calendar(month, year)
+          break
+        else
+          puts "Invalid date".red
+        end
+        break
+      rescue => e
+        #puts e.message
+        #p e.backtrace.join("\n")
+        puts "Invalid input".red
+        puts ""
+      end
+    end
+  end
+
+  def add_event
+    date = validate_date
+    title = nil
+    desc = nil
+    venue = nil
+  loop do
+    begin
+        print "Enter title: "
+        title = gets.chomp
+        if title != nil && title != ""
+          break
+        end
+      rescue
+        puts "Invalid Input"
+      end
+    end
+    print "Enter description: "
+    desc = gets.chomp
+    print "Enter venue: "
+    venue = gets.chomp
+
+    event_object = EventInfo.new(title,desc,venue)
+    if @all_events.has_key? date
+      @all_events[date].push event_object
+    else
+      @all_events[date] = [event_object]
+    end
+    puts "Event added successfuly".green
+    puts ""
+  end
+
+  def validate_date
+    day = 0
+    month = 0
+    year = 0
+    loop do
+      loop do
+        begin
+          print "Enter year: "
+          year = Integer(gets)
+          if year != nil && year != 0 && year > 1500 && year < 3000
+            break
+          else
+            puts "Year can be between 1500 to 3000".yellow
+          end
+        rescue
+          puts "Invalid Year".red
+        end
+      end
+
+      loop do
+        begin
+          print "Enter month: "
+          month = Integer(gets)
+          if month != nil && month != 0 && month >= 1 && month <= 12
+            break
+          else
+            puts "Month value can be between 1 to 12".yellow
+          end
+        rescue
+          puts "Invalid Month".red
+        end
+      end
+
+      loop do
+        begin
+          print "Enter day: "
+          day = Integer(gets)
+          if day != nil && day != 0 && day >= 1 && day <= 31
+            break
+          else
+            puts "Day value can be between 1 to 31".yellow
+          end
+        rescue
+          puts "Invalid Day".red
+        end
+      end
+
+      if Date.valid_date?(year, month, day)
+        break      
+      else
+        puts "Invalid date".yellow
+      end
+    end
+    "#{day}-#{month}-#{year}"
```

Comment:
a method that is supposed to return a date should actually return a date object and not a string.